src/System.Net.Security/src/System/Net/Security/SslStreamInternal.Adapters.cs

```diff
@@ -15,6 +15,39 @@ private interface ISslWriteAdapter
             Task WriteAsync(byte[] buffer, int offset, int count);
         }
 
+        private interface ISslReadAdapter
+        {
+            ValueTask<int> ReadAsync(byte[] buffer, int offset, int count);
+            ValueTask<int> LockAsync(byte[] buffer, int offset, int count);
+        }
+
+        private struct SslReadAsync : ISslReadAdapter
+        {
+            private readonly SslState _sslState;
+            private readonly CancellationToken _cancellationToken;
+
+            public SslReadAsync(SslState sslState, CancellationToken cancellationToken)
+            {
+                _cancellationToken = cancellationToken;
+                _sslState = sslState;
+            }
+
+            public ValueTask<int> ReadAsync(byte[] buffer, int offset, int count) => new ValueTask<int>(_sslState.InnerStream.ReadAsync(buffer, offset, count, _cancellationToken));
```

Comment:
Can you use the Memory overload here? If it completes synchronously, that'll save a task allocation.

Reply:
Done.

Reply:
> Done

This is still calling the array overload.  Isn't this just wrapping a Task in a ValueTask?